src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -1352,10 +1584,16 @@ def get_connected_passed_kwargs(prefix):
                 f"Pipeline {pipeline_class} expected {expected_modules}, but only {passed_modules} were passed."
             )
 
-        # 8. Instantiate the pipeline
+        # 10. Instantiate the pipeline
         model = pipeline_class(**init_kwargs)
 
-        # 9. Save where the model was instantiated from
+        # 11. Handle cases where device map strategy is not "balanced".
+        if device_map == "balanced_ultra_low_memory":
+            model.enable_sequential_cpu_offload(component_device_mapping=final_device_map)
+        elif device_map == "balanced_low_memory":
+            model.enable_model_cpu_offload(component_device_mapping=final_device_map)
```

Comment:
Decided to instead leverage the existing functionalities. This should suffice no?